### PR TITLE
reformat conv3d

### LINKED ISSUE #7390: conv3d

### PR DESCRIPTION
**Pull request by NouranYoussef**
Closes #7390

### CODE REVIEW

**Comment by NouranYoussef** [comment]
Close #7390

File: ivy/functional/backends/tensorflow/layers.py [hunk 1]
```diff
@@ -173,14 +173,14 @@ def depthwise_conv2d(
 @with_unsupported_devices({"2.9.1 and below": ("cpu",)}, backend_version)
 # noinspection PyDefaultArgument
 def conv3d(
-    x,
-    filters,
-    strides,
-    padding,
+    x: Union[tf.Tensor, tf.Variable],
```
Comment (kurshakuz) [inline ivy/functional/backends/tensorflow/layers.py:176]: I think the conv3d() method can only accept Tensors as x and filters. Please check this out: https://www.tensorflow.org/api_docs/python/tf/nn/conv3d

File: ivy/functional/ivy/layers.py [hunk 2]
```diff
@@ -1217,12 +1217,12 @@ def depthwise_conv2d(
 def conv3d(
     x: Union[ivy.Array, ivy.NativeArray],
```
Comment (kurshakuz) [inline ivy/functional/ivy/layers.py:1218]: I see that in the docstring examples you used Container classes as inputs. If the function can accept them, then we need to add them here as well.

File: ivy/functional/ivy/layers.py [hunk 3]
```diff
@@ -1217,12 +1217,12 @@ def depthwise_conv2d(
...
     filters: Union[ivy.Array, ivy.NativeArray],
```
Comment (kurshakuz) [inline ivy/functional/ivy/layers.py:1219]: Similarly here

File: ivy/functional/ivy/layers.py [hunk 4]
```diff
@@ -1217,12 +1217,12 @@ def depthwise_conv2d(
...
-    strides: int,
+    strides: Union[int, Tuple[int, int, int]],
```
Comment (kurshakuz) [inline ivy/functional/ivy/layers.py:1220]: I am not totally sure, but can this be: `strides: Union[int, Tuple[int], Tuple[int, int], Tuple[int, int, int]]`? I noticed similar pattern in the Conv2d implementation.

File: ivy/functional/ivy/layers.py [hunk 5]
```diff
@@ -1217,12 +1217,12 @@ def depthwise_conv2d(
...
     padding: str,
     /,
     *,
     data_format: str = "NDHWC",
-    dilations: int = 1,
+    dilations: Optional[Union[int, Tuple[int, int, int]]] = 1,
```
Comment (kurshakuz) [inline ivy/functional/ivy/layers.py:1225]: Similarly here

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Hey! This is looking very good for me! Can you please check the last comments I left? After figuring them out, I think it would be good to merge. Thanks 🙂

File: ivy/functional/ivy/layers.py (hunk 4 above)
Comment (NouranYoussef) [reply ivy/functional/ivy/layers.py]: updated now

File: ivy/functional/ivy/layers.py (hunk 3 above)
Comment (NouranYoussef) [reply ivy/functional/ivy/layers.py:1219]: yeah it is noted in the docstring that the function is nestable and also accepts container classes as noted in the guide

File: ivy/functional/backends/tensorflow/layers.py (hunk 1 above)
Comment (NouranYoussef) [reply ivy/functional/backends/tensorflow/layers.py:176]: I believe tensorflow allows tf.variable to be used wherever tensors are. let me know if change is still required

File: ivy/functional/ivy/layers.py (hunk 3 above)
Comment (kurshakuz) [reply ivy/functional/ivy/layers.py:1219]: Can you add them here as inputs as well for completeness?

File: ivy/functional/ivy/layers.py (hunk 4 above)
Comment (kurshakuz) [reply ivy/functional/ivy/layers.py]: Sorry for back and forth, but I checked the TF and Torch documentation again and it seems like the initial idea was correct. Can I please ask you to try and run a couple of tests and see whether the function can accept these types of inputs. Thanks!

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Looking good to me already. Can you please check last two comments? After resolving them the PR is good to go.

File: ivy/functional/ivy/layers.py (hunk 4 above)
Comment (NouranYoussef) [reply ivy/functional/ivy/layers.py:1220]: it gives "tuple index out of range " as tuples dim3 is accessed in the backend functions, i edited it back as in the docs .idk if that pattern was desired in ivy , the backend funcs should be modified

File: ivy/functional/ivy/layers.py (hunk 3 above)
Comment (NouranYoussef) [reply ivy/functional/ivy/layers.py]: done

**Review by kurshakuz — APPROVED** [review]
LGTM! Thanks for your contribution 🙂

**Pull request merged**